### PR TITLE
Esx support with possibility for qb

### PR DESCRIPTION
**Pull request by Sampuke**

### CODE REVIEW

**Comment by Sampuke** [comment]
I also added Config.Performance if some people have mechanic jobs which are used only for performance upgrades.

**Comment by alberttheprince** [comment]
@Jorn08 please review big homie

File: server.lua [hunk 1]
```diff
@@ -1,8 +1,57 @@
...
+    Core = exports['qb-core']:GetCoreObject()
+    CoreName = 'qb'
+elseif GetResourceState('es_extended') == 'started' then
+    Core = exports["es_extended"]:getSharedObject()
+    CoreName = 'esx'
 else
-    warn('qb-core is missing, modifications won\'t cost anything')
+    print('Core is missing nothing will cost')
+end
+Functions = {}
```
Comment (Jorn08) [inline server.lua:11]: Is there a reason this should be global?

File: client/menus/performance.lua [hunk 2]
```diff
@@ -1,3 +1,4 @@
+if not Config.Performance then return end
```
Comment (Jorn08) [inline client/menus/performance.lua:1]: Already prevented by `if Config.Performance then` in main.lua#L54?

File: client/menus/main.lua [hunk 3]
```diff
@@ -138,7 +145,8 @@ menu.onClose = function()
 end
 
 lib.callback.register('customs:client:vehicleProps', function()
-    return QBCore.Functions.GetVehicleProperties(vehicle)
+    return lib.getVehicleProperties(vehicle)
```
Comment (Jorn08) [inline client/menus/main.lua:148]: qb uses [`QBCore.Functions.SetVehicleProperties`](https://docs.qbcore.org/qbcore-documentation/qb-core/client-function-reference#qbcore.functions.setvehicleproperties)  to set the vehicle properties back, will this not break that?

File: client/menus/main.lua [hunk 4]
```diff
@@ -1,14 +1,20 @@
 mainLastIndex = 1
 vehicle = 0
 mainMenuId = 'customs-main'
-local QBCore
+local Core, corename
```
Comment (Jorn08) [inline client/menus/main.lua:4]: This is unused

File: client/menus/main.lua [hunk 5]
```diff
@@ -1,14 +1,20 @@
...
 
 if GetResourceState('qb-core') == 'started' then
-    QBCore = exports['qb-core']:GetCoreObject()
+    Core = exports['qb-core']:GetCoreObject()
+    corename = 'qb'
+elseif GetResourceState('es_extended') == 'starter' then
+    Core = exports['es_extended']:getSharedObject()
+    corename = 'esx'
+else
+    print('You arent using esx or qb, if you want payment to work use one of them.')
```
Comment (Jorn08) [inline client/menus/main.lua:17]: Making this code redundant

**Review by Jorn08 — CHANGES_REQUESTED** [review]
I would prefer the framework specified client and server functions to be moved in their own files. e.g. `client/core.lua`, `server/core.lua`, or a single file with distinction between client and server.

File: client/menus/main.lua [hunk 6]
```diff
@@ -1,14 +1,20 @@
...
 local inMenu = false
 local dragcam = require('client.dragcam')
 local startDragCam = dragcam.startDragCam
 local stopDragCam = dragcam.stopDragCam
 
 if GetResourceState('qb-core') == 'started' then
-    QBCore = exports['qb-core']:GetCoreObject()
+    Core = exports['qb-core']:GetCoreObject()
+    corename = 'qb'
+elseif GetResourceState('es_extended') == 'starter' then
```
Comment (mahanmoulaei) [inline client/menus/main.lua:13]: There is a typo here

File: client/zones.lua [hunk 7]
```diff
@@ -1,11 +1,35 @@
 local zoneId
-local QBCore
+local Core, CoreName
 local allowAccess = false
 
 if GetResourceState('qb-core') == 'started' then
-    QBCore = exports['qb-core']:GetCoreObject()
+    Core = exports['qb-core']:GetCoreObject()
+    CoreName = 'qb'
+elseif GetResourceState('es_extended') == 'starter' then
```
Comment (mahanmoulaei) [inline client/zones.lua:8]: There is a typo here

File: server.lua [hunk 8]
```diff
@@ -18,16 +67,16 @@ end
...
+    local cashBalance = Functions[CoreName].GetCash(player)
+    local bankBalance = Functions[CoreName].GetBank(player)
 
     if cashBalance >= amount then
-        player.Functions.RemoveMoney('cash', amount, "Customs")
+        Functions[CoreName].RemoveMoney(player, 'cash', amount)
         return true
     elseif bankBalance >= amount then
-        player.Functions.RemoveMoney('bank', amount, "Customs")
+        Functions[CoreName].RemoveMoney(player, 'bank', amount)
```
Comment (mahanmoulaei) [inline server.lua:79]: If I'm not mistaken, both qb and esx support reason parameter to add/remove money so you can keep it there

File: server.lua [hunk 9]
```diff
@@ -1,8 +1,57 @@
...
+    return player.getMoney()
+end
+Functions.esx.GetBank = function(player)
+    return player.getAccount('bank').money
+end
+Functions.esx.RemoveMoney = function(player, type, amount)
+    return player.removeAccountMoney(type, amount)
+end
+Functions.esx.GetJob = function(player)
+    return player.getJob().name
```
Comment (mahanmoulaei) [inline server.lua:47]: It's a good practice to use the safe operator of `?` for nil checking to reduce code error. Same thing applies to GetBank()

File: server.lua [hunk 10]
```diff
@@ -1,8 +1,57 @@
...
+    return Core.Functions.GetPlayer(src)
+end
+Functions.qb.GetMoney = function(player, type)
+    return player.Functions.GetMoney(type)
+end
+Functions.qb.RemoveMoney = function(player, type, amount)
+    return player.Functions.RemoveMoney(type, amount, "Customs")
+end
+Functions.qb.GetJob = function(player)
+    return player.PlayerData.job.name
```
Comment (mahanmoulaei) [inline server.lua:24]: It's a good practice to use the safe operator of `?` for nil checking to reduce code error

**Pull request closed without merging**